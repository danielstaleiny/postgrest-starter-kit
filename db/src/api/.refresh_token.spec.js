const ROOT = process.cwd()
const { get, post, put, patch, del } = require(`${ROOT}/util/fetch.js`)(
  'http://localhost:3000/'
)

describe('/rpc/refresh_token', () => {
  test('fail timed out 401 Session time out', async () => {
    const res = await post('rpc/refresh_token')({
      body: {
        refresh_token: 'de688895-6181-440f-a25a-8a9ba14b2163',
      },
    })
    expect(res).toMatchSnapshot()
  })

  test('200 get new token', async () => {
    const res = await post('rpc/refresh_token')({
      body: {
        refresh_token: 'de688895-6181-440f-a25a-8a9ba14b2164',
      },
    })
    res.body.token = 'checked'
    expect(res).toMatchSnapshot()
  })

  test('404 Session not found', async () => {
    const res = await post('rpc/refresh_token')({
      body: {
        refresh_token: 'de688895-6181-440f-a25a-000000000000',
      },
    })
    expect(res).toMatchSnapshot()
  })

  test('404 CSRF not found', async () => {
    const res = await post('rpc/refresh_token')({
      body: {
        refresh_token: 'de688895-6181-440f-a25a-8a9ba14b2164',
        csrf: 'de6888956181440fa25a000000000000',
      },
    })
    expect(res).toMatchSnapshot()
  })
})
