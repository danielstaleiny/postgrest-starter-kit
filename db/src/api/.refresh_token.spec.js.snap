// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`/rpc/refresh_token 200 get new token 1`] = `
Object {
  "body": Object {
    "token": "checked",
  },
  "status": 200,
  "statusText": "OK",
}
`;

exports[`/rpc/refresh_token 404 CSRF not found 1`] = `
Object {
  "body": Object {
    "details": "CSRF not found",
    "hint": "",
  },
  "status": 404,
  "statusText": "Not Found",
}
`;

exports[`/rpc/refresh_token 404 Session not found 1`] = `
Object {
  "body": Object {
    "details": "Session not found",
    "hint": "Login again to receive new valid refresh_token",
  },
  "status": 404,
  "statusText": "Not Found",
}
`;

exports[`/rpc/refresh_token fail timed out 401 Session time out 1`] = `
Object {
  "body": Object {
    "details": "Session expired",
    "hint": "Login again to receive new valid refresh_token",
  },
  "status": 401,
  "statusText": "Unauthorized",
}
`;
